tests/ui/lint/unreachable_pub.stderr

```diff
@@ -1,140 +1,270 @@
 warning: unreachable `pub` item
-  --> $DIR/unreachable_pub.rs:8:13
+  --> $DIR/unreachable_pub.rs:10:13
    |
 LL |     pub use std::fmt;
    |     ---     ^^^^^^^^
    |     |
-   |     help: consider restricting its visibility: `pub(crate)`
+   |     help: consider restricting its visibility: `pub(super)`
```

Comment:
If `pub(super) == pub(crate)` then `pub(crate)` should be recommended.

Reply:
I'm not following, when is `pub(super)` equal to `pub(crate)` ?

Is this what you mean?
```
mod private_mod {
    pub use std::fmt; // should be `pub(crate)`

    mod inner_mod {
        pub use std::fmt; // should be `pub(super)`
    }
}
```

Reply:
Yes.

Reply:
I find this a bit odd, what's the reasoning for recommending `pub(crate)` in those cases instead of `pub(super)`?

Reply:
That's what people typically do when restricting visibilities manually.
If you see `pub(crate)` you know that the item is visible to the whole crate, without knowing how deep in the modules it is.

Reply:
That makes sense. I have adjusted the handling to take into account `super` being the crate root.